### PR TITLE
Add container option when link a cluster in Eclipse

### PR DESCRIPTION
**Pull request by t-rufang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java [hunk 1]
```diff
@@ -65,6 +73,19 @@ protected void configureShell(Shell newShell) {
...
     }
 
+    private void refreshContainers(@NotNull ClientStorageAccount storageAccount) {
+    	try {
+        	containersComboBox.removeAll();
+
+			StorageClientSDKManager.getManager().getBlobContainers(storageAccount.getConnectionString())
+					.forEach(blob -> containersComboBox.add(blob.getName()));
+			
+			// not find setMaximumRowCount(*) method in SWT
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: Just make sure that the combobox UI is working

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java [hunk 2]
```diff
@@ -105,7 +126,7 @@ protected Control createDialogArea(Composite parent) {
         gridData.grabExcessHorizontalSpace = true;
         storageNameField.setLayoutData(gridData);
         storageNameField.setToolTipText("The default storage account of the HDInsight cluster, which can be found from HDInsight cluster properties of Azure portal.\n\n Press the F1 key or click the '?'(Help) button to get more details");
-
+
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: white space

Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: spaces removed.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java (hunk 1 above)
Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: Combobox UI does work in my tests.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java [hunk 3]
```diff
@@ -65,6 +75,19 @@ protected void configureShell(Shell newShell) {
...
+    private void refreshContainers(@NotNull ClientStorageAccount storageAccount) {
+    	try {
+        	containersComboBox.removeAll();
+
+			StorageClientSDKManager.getManager().getBlobContainers(storageAccount.getConnectionString())
+					.forEach(blob -> containersComboBox.add(blob.getName()));
+			
+			// not find setMaximumRowCount(*) method in SWT
+		} catch (AzureCmdException e) {
+			containersComboBox.removeAll();
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: >containersComboBox.removeAll(); [](start = 3, length = 31)

Probably we should not removeAll? What's wrong with the containers that are already fetched.

Comment (djlan) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: Recommend to delete "removeAll" and add error message here

---
In reply to: [164978754](https://github.com/Microsoft/azure-tools-for-java/pull/1360#discussion_r164978754) [](ancestors = 164978754)

Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: "removeAll" is deleted and error message is added.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java [hunk 4]
```diff
@@ -62,9 +72,24 @@ public AddNewClusterForm(Shell parentShell, HDInsightRootModule module) {
...
+    private void refreshContainers(@NotNull ClientStorageAccount storageAccount) {
+    	try {
+        	containersComboBox.removeAll();
+
+			StorageClientSDKManager.getManager().getBlobContainers(storageAccount.getConnectionString())
+					.forEach(blob -> containersComboBox.add(blob.getName()));
+			
+			// not find setMaximumRowCount(*) method in SWT
+		} catch (AzureCmdException e) {
+			setErrorMessage(e.getMessage());
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: What's the error message if the account/key is wrong?

Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: I test this case and error message is "Error retrieving the Blob Container List".

Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/AddNewClusterForm.java]: The container `empty check` is already done in Line 291.

**Pull request merged**